ui/models.py

```diff
@@ -42,45 +44,104 @@ class Video(models.Model):
     description = models.TextField(blank=True)
     source_url = models.URLField()
     moira_lists = models.ManyToManyField(MoiraList)
+    status = models.TextField(max_length=24, blank=True)
+    s3_subkey = models.UUIDField(unique=True, null=False, blank=False, default=uuid4)
+
+    @property
+    def s3_key(self):
+        """
+        Avoid duplicate S3 keys/filenames when transferring videos rom Dropbox
+
+        Args:
+            user_id (str): The user who is uploading a new video
+            key (str): The base key to check for duplicates
+
+        Returns:
+            str: A unique S3 key including the user id as a virtual subfolder
+        """
+        basename, extension = os.path.splitext(self.source_url.split('/')[-1])
+        newkey = '{user}/{uuid}/{base}{ext}'.format(
+            user=self.creator.id, uuid=str(self.s3_subkey), base=basename, ext=extension)
+        return newkey
+
+
+class VideoFile(models.Model):
+    """
+    A file associated with a Video object, either the original upload or a transcoded file.
+    """
+    created_at = models.DateTimeField(auto_now_add=True)
+    s3_object_key = models.TextField(unique=True)
+    bucket_name = models.CharField(max_length=128, blank=False, null=False)
+    video = models.ForeignKey(Video, on_delete=models.CASCADE)
+    encoding = models.CharField(max_length=128, default='original')
+    preset_id = models.CharField(max_length=128, blank=True, null=True)
```

Comment:
preset_id will be the AWS encoding preset used for the VideoFile, not used here yet.  I can take it out from this PR but it will be needed for the transcoding PR.